tests/std/tests/P0896R4_stream_iterators/test.cpp

```diff
@@ -2,15 +2,21 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 // Covers Ranges changes to istream_iterator, ostream_iterator, istreambuf_iterator, and ostreambuf_iterator
+// and LWG-3719 changes for directory_iterator, recursive_directory_iterator, regex_iterator, and regex_token_iterator
 
 #include <cassert>
 #include <cstddef>
 #include <iostream>
 #include <iterator>
+#include <regex>
 #include <string>
 #include <type_traits>
 #include <utility>
 
+#if _HAS_CXX17

```

Comment:
I'm pretty certain `_HAS_CXX17` is always true in a Ranges test =)

Reply:
This test is a bit strange (pre-existing). While being a Ranges test, it uses `usual_matrix.lst`.

https://github.com/microsoft/STL/blob/04ee87830a404437f1691d6a3e8809e4986d8493/tests/std/tests/P0896R4_stream_iterators/env.lst#L1-L4

Should we use `concepts_20_matrix.lst` instead?

Reply:
I stand corrected! I'd forgotten that this test runs in older modes to make sure that the conditional changes from C++20 didn't break the subject iterators in older language modes. Let's leave it with `usual_matrix.lst`.